sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -97,6 +97,7 @@ private static void CheckCountCall(SyntaxNodeAnalysisContext context)
             var invocation = (InvocationExpressionSyntax)context.Node;
             if (invocation.ArgumentList?.Arguments.Count == 0 &&
                 invocation.Expression is MemberAccessExpressionSyntax memberAccess &&
+                invocation.ToStringContains(CountName) &&
```

Comment:
the perf improvements are not visible, mainly because this rule checks multiple calls, not just the Count calls

Reply:
it just went down from ~11710 to ~11554

Reply:
All three `CheckExtensionMethodsOnIEnumerable`, `CheckToCollectionCalls` and `CheckCountCall` needs similar improvement to see the difference.

While there should be some SemanticsModel cache on Roslyn side, updating one of them should have low effect alone on the numbers. Let's keep it as a first babystep for #3731